### ISSUE TITLE
"end" event not triggered, not pulling data out of stream?

### ISSUE THREAD

**Issue by fkling** [deoxxa/duplexer2#15]
Since upgrading to 0.1.0, my code didn't work anymore as intended. I think it's because the internals changed and `.resume` is not called on the readable stream anymore.

I spent quite a while to figure out whether issue is in my (rather complex) code, but still wasn't able to get it to work.
So I tried to create a minimal example to replicate the issue. It seems the issue can be surfaced by having an asynchronous process in `_write` of the writable side:

```javascript
var through2 = require("through2");

var duplexer2 = require("./");
var stream = require("readable-stream");


function getMainStream() {
  var writable = new stream.Writable({objectMode: true});
  var readable = through2.obj();

  writable._write = function(data, _, next) {
    setTimeout(() => {
      console.log('written', data);
      readable.write(data);
      next();
    }, 0);
  };
  writable.once('finish', () => {
    console.log('finish');
    readable.end();
  });

  return duplexer2({objectMode: true}, writable, readable);
}

var s = getMainStream()
  .once('end', () => console.log('done'))
  .on('data', d => console.log('out', d));
s.write('a');
s.end();
```

This is creating a stream and writing one value to it. The stream is "opened" by listing to the `data` event, but the same behavior can be observed when listing to `readable` instead or calling `resume`.

The write side receives data and simply passing it along to the readable side (which is a passthrough stream). Once writing is done (`finish` event), the readable side is closed (`readable.end()`).

When I run this code, this is the output I get:

```
$ node myTest.js
written a
out a
finish
```

As you can see, the `end` event triggered. If I comment out the `setTimeout` part, it works as expected.

**Note:** If I return the `readable` and `writable` directly (e.g. `return [readable, writable];`), instead of using `duplexer2`, attach the event handlers to the readable side and write to writable, I get the expected output.

Is there an issue with duplexer2 or am I just misunderstanding streams, in which case, how should this be done correctly (and sorry for hijacking this project for my problem)?

**Comment by shinnn** [comment]
Yes, confirmed. It's a bug.

I'll solve this problem in a day. Thanks for the detailed report.

**Comment by shinnn** [comment]
Though fixed in v0.1.1, I still looking for a smarter solution, just reverting `data` event based code. https://github.com/deoxxa/duplexer2/pull/6#discussion_r42305262
